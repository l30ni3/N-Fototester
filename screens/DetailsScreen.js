import React, {useState, useEffect} from 'react';
import {StyleSheet, SafeAreaView, View, Text, Image} from 'react-native';
import {
  Button,
  Divider,
  Icon,
  Layout,
  TopNavigation,
  TopNavigationAction,
} from '@ui-kitten/components';

const BackIcon = props => <Icon {...props} name="arrow-back" />;

export const DetailsScreen = ({route, navigation}) => {
  const {itemId} = route.params;
  const [data, setData] = useState([]);
  const navigateBack = () => {
    navigation.goBack();
  };

  const BackAction = () => (
    <TopNavigationAction icon={BackIcon} onPress={navigateBack} />
  );

  useEffect(() => {
    fetch(`http://localhost:5000/api/images/${itemId}`)
      .then(res => res.json())
      .then(json => setData(json))
      .catch(error => console.error(error));
  }, []);

  useEffect(() => {
    console.log(data.name);
  }, [data]);

  return (
    <SafeAreaView style={{flex: 1}}>
      <TopNavigation
        title="Details"
        alignment="center"
        accessoryLeft={BackAction}
      />
      <Divider />
      <Layout
        style={{
          flex: 1,
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        {data ? (
          <>
            <Text>Daten:</Text>
            <View style={styles.container}>
              {/* <Image source={{uri: photo.uri}} style={styles.preview} /> */}
            </View>
          </>
        ) : (
          <Text>Keine Daten</Text>
        )}
      </Layout>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'column',
    backgroundColor: 'black',
  },
  preview: {
    flex: 1,
    justifyContent: 'center',
  },
  button: {
    margin: 20,
  },
});
