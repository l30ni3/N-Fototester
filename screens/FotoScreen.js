import React from 'react';
import {SafeAreaView} from 'react-native';
import {
  Button,
  Divider,
  Icon,
  Layout,
  Text,
  TopNavigation,
  TopNavigationAction,
} from '@ui-kitten/components';

const BackIcon = props => <Icon {...props} name="arrow-back" />;

export const FotoScreen = ({navigation}) => {
  const navigateBack = () => {
    navigation.goBack();
  };

  const BackAction = () => (
    <TopNavigationAction icon={BackIcon} onPress={navigateBack} />
  );

  return (
    <SafeAreaView style={{flex: 1}}>
      <TopNavigation
        title="Neue Messung"
        alignment="center"
        accessoryLeft={BackAction}
      />
      <Divider />
      <Layout
        style={{
          flex: 1,
        }}></Layout>
    </SafeAreaView>
  );
};
