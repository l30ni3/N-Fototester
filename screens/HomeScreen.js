import React, {useState, useEffect} from 'react';
import {
  Avatar,
  Button,
  Icon,
  Divider,
  Layout,
  List,
  ListItem,
  Text,
  TopNavigation,
  TopNavigationAction,
} from '@ui-kitten/components';
import {SafeAreaView} from 'react-native';
import {useIsFocused} from '@react-navigation/native';
import Moment from 'react-moment';
import {fetchResults, deleteResult, fetchImage} from './services';
const GLOBAL = require('./components/constants');

export const HomeScreen = props => {
  const navigation = props.navigation;
  const [data, setData] = useState([]);
  const [avatars, setAvatars] = useState([]);
  const MenuIcon = props => <Icon {...props} name="menu-outline" />;

  const isFocused = useIsFocused();

  const MenuAction = () => (
    <TopNavigationAction
      icon={MenuIcon}
      onPress={() => navigation.openDrawer()}
    />
  );

  const renderAvatar = props => {
    var avatar = avatars.reduce((result, item) => {
      if (item.name === props.name) {
        result = `${result}${item.uri}`;
      }
      return result;
    }, '');

    console.log(avatar);

    return (
      <Avatar
        size="tiny"
        source={
          avatar ? {uri: avatar} : {uri: 'https://www.gravatar.com/avatar'}
        }
      />
    );
  };

  const renderItemAccessory = props => (
    <>
      <Button
        size="tiny"
        onPress={() => {
          navigation.navigate('Ergebnis', {
            itemId: props.itemId,
          });
        }}>
        Anzeigen
      </Button>
      <Icon
        {...props}
        onPress={() => {
          console.log(props.itemId);
          deleteResult({...props}).then(res => setData(res));
        }}
        name="trash-outline"
      />
    </>
  );

  const renderItem = ({item, index}) => (
    <ListItem
      accessoryLeft={props => renderAvatar({...{name: item.name}, ...props})}
      title={
        <Moment element={Text} fromNow>
          {item.date}
        </Moment>
      }
      accessoryRight={props =>
        renderItemAccessory({...{itemId: item.id}, ...props})
      }
    />
  );

  useEffect(() => {
    fetchResults().then(json => setData(json));
  }, []);

  //when using react navigation, screens need to refresh also on isFocused event
  useEffect(() => {
    fetchResults().then(json => setData(json));
  }, [isFocused]);

  useEffect(() => {
    const updatedAvatars = [];
    data.map(item =>
      updatedAvatars.push({
        name: item.name,
        uri: fetchImage(item.name),
      }),
    );
    setAvatars(updatedAvatars);
  }, [data]);

  useEffect(() => {
    console.log('avatars', avatars);
  }, [avatars]);

  return (
    <SafeAreaView style={{flex: 1}}>
      <TopNavigation
        title="Meine Messungen"
        alignment="center"
        accessoryLeft={MenuAction}
      />
      <Divider />
      <Layout
        style={{
          flex: 1,
        }}>
        <List data={data} renderItem={renderItem} />
        <Button
          style={{
            margin: 30,
          }}
          title="Neue Messung starten"
          testID="buttonOnPress"
          onPress={() => navigation.navigate('Tutorial 1')}>
          Neue Messung starten
        </Button>
      </Layout>
    </SafeAreaView>
  );
};
