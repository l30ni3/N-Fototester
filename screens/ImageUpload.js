import React, {useState, useRef, useEffect} from 'react';
import {SafeAreaView, StyleSheet, View, Image, Platform} from 'react-native';
import {
  Button,
  Divider,
  Icon,
  Layout,
  TopNavigation,
  TopNavigationAction,
  Spinner,
  Text,
} from '@ui-kitten/components';
import {launchCamera, launchImageLibrary} from 'react-native-image-picker';
import {RNCamera} from 'react-native-camera';
import Permissions from 'react-native-permissions';

const GLOBAL = require('./components/constants');

export const ImageUpload = ({navigation}) => {
  const [photo, setPhoto] = useState(null);
  const [isLoading, setIsLoading] = useState(false);
  const [itemId, setItemId] = useState(null);
  const [permission, setPermission] = useState('undetermined');

  const BackIcon = props => <Icon {...props} name="chevron-left-outline" />;
  const CameraIcon = props => <Icon {...props} name="camera" />;
  const UploadIcon = props => <Icon {...props} name="upload" />;
  const CloseIcon = props => <Icon {...props} name="close" />;
  const NextIcon = props => (
    <Icon {...props} name="arrow-ios-forward-outline" />
  );

  const navigateBack = () => {
    navigation.goBack();
  };

  const BackAction = () => (
    <TopNavigationAction icon={BackIcon} onPress={navigateBack} />
  );

  useEffect(() => {
    Permissions.check('photo').then(response => {
      // Response is one of: 'authorized', 'denied', 'restricted', or 'undetermined'
      setPermission(response);
    });
  }, []);

  useEffect(() => {
    itemId
      ? navigation.navigate('Ergebnis', {
          itemId: itemId,
        })
      : null;
  }, [itemId]);

  const handleTakePhoto = () => {
    launchCamera({noData: true}, response => {
      if (response) {
        setPhoto(response);
      }
    });
  };

  const handleChoosePhoto = () => {
    const options = {
      mediaType: 'photo',
      includeBase64: true,
    };
    launchImageLibrary({options}, response => {
      if (response.didCancel) {
        console.log('User cancelled image picker');
        setPhoto(null);
      } else if (response.error) {
        console.log('ImagePicker Error: ', response.error);
        setPhoto(null);
      } else {
        setPhoto(response);
      }
    });
  };

  const handleDeletePhoto = () => {
    setPhoto(null);
  };

  const handleUploadPhoto = () => {
    setIsLoading(true);
    let localUri =
      Platform.OS === 'ios' ? photo.uri.replace('file://', '') : photo.uri;
    let filename = photo.fileName;
    let match = /\.(\w+)$/.exec(filename);
    let type = match ? `image/${match[1]}` : `image`;
    let formData = new FormData();
    formData.append('photo', {
      uri: localUri,
      name: filename,
      type: type,
    });
    fetch(`${GLOBAL.SERVER_URL}/api/upload`, {
      method: 'POST',
      body: formData,
      // headers: {
      //   Accept: 'application/json',
      //   'Content-Type': 'multipart/form-data',
      // },
    })
      .then(response => response.text())
      .then(response => {
        console.log('upload succes', response);
        setItemId(response);
        setIsLoading(false);
        setPhoto(null);
      })
      .catch(error => {
        console.log('upload error', error);
        setIsLoading(false);
      });
  };

  return (
    <SafeAreaView style={{flex: 1}}>
      <TopNavigation
        title="Neue Messung"
        alignment="center"
        accessoryLeft={BackAction}
      />
      <Divider />
      <Layout
        style={{
          flex: 1,
        }}>
        {!isLoading ? (
          <>
            {photo ? (
              <View style={styles.container}>
                <Image source={{uri: photo.uri}} style={styles.preview} />
              </View>
            ) : (
              <View style={styles.container}></View>
            )}
            <View
              style={{
                flexDirection: 'row',
                justifyContent: 'center',
              }}>
              {photo ? (
                <>
                  <Button
                    style={styles.button}
                    onPress={handleDeletePhoto}
                    accessoryRight={CloseIcon}>
                    Verwerfen
                  </Button>
                  <Button
                    style={styles.button}
                    onPress={handleUploadPhoto}
                    accessoryRight={NextIcon}>
                    Verwenden
                  </Button>
                </>
              ) : (
                <>
                  <Button
                    style={styles.button}
                    onPress={handleChoosePhoto}
                    accessoryRight={UploadIcon}>
                    Aus Gallerie
                  </Button>
                  <Button
                    style={styles.button}
                    onPress={handleTakePhoto}
                    accessoryRight={CameraIcon}>
                    Neues Foto
                  </Button>
                </>
              )}
            </View>
          </>
        ) : (
          <View style={styles.spinner}>
            <Spinner size="large" />
            <Text style={styles.spinnertext}>
              Die Bilddaten werden verarbeitet ...{' '}
            </Text>
          </View>
        )}
      </Layout>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'column',
    backgroundColor: 'black',
  },
  preview: {
    flex: 1,
    justifyContent: 'center',
  },
  button: {
    margin: 20,
  },
  spinner: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  spinnertext: {
    color: 'white',
    margin: 20,
  },
});
