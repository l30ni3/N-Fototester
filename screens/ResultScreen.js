import React, {useState, useRef, useEffect} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  View,
  Image,
  Platform,
  Text,
} from 'react-native';
import {
  Avatar,
  Button,
  Divider,
  Icon,
  Layout,
  TopNavigation,
  TopNavigationAction,
  Spinner,
} from '@ui-kitten/components';
import {useNavigation} from '@react-navigation/native';

const SERVER_URL = 'http://localhost:5000';

export const ResultScreen = ({route, navigation}) => {
  // const navigation = useNavigation();
  const {itemId} = route.params;
  const [isLoading, setIsLoading] = useState(false);
  const [data, setData] = useState([]);
  const [avatar, setAvatar] = useState('');

  const BackIcon = props => <Icon {...props} name="chevron-left-outline" />;

  const navigateBack = () => {
    navigation.navigate('Meine Messungen');
  };

  const BackAction = () => (
    <TopNavigationAction icon={BackIcon} onPress={navigateBack} />
  );

  // const renderAvatar = props => require(`../api/avatars/${props.name}`);

  useEffect(() => {
    fetch(`http://localhost:5000/api/results/${itemId}`)
      .then(res => res.json())
      .then(json => setData(json))
      .catch(error => console.error(error));
  }, []);

  useEffect(() => {
    setAvatar(`../api/avatars/${data.name}`);
    if (data) {
      console.log(avatar);
    }
  }, [data]);

  return (
    <SafeAreaView style={{flex: 1}}>
      <TopNavigation
        title="Ergebnis"
        alignment="center"
        accessoryLeft={BackAction}
      />
      <Divider />
      <Layout
        style={{
          flex: 1,
        }}>
        <View style={styles.image_container}>
          {data && (
            // <Avatar
            //   style={styles.avatar}
            //   size="giant"
            //   source={require('../api/avatars/728EE956-850C-43D7-8551-0421D7063611.jpg')}
            // />
            <Image
              style={styles.avatar}
              source={require('../api/avatars/728EE956-850C-43D7-8551-0421D7063611.jpg')}
            />
            // <Image style={styles.avatar} source={require(`${avatar}`)} />
          )}
        </View>
        {!isLoading ? (
          <View style={styles.container}>
            <Text style={styles.text_bold}>Hue Circular Mean</Text>
            <Text style={styles.text}>{data.hue_circular_mean}</Text>
            <Text style={styles.text_bold}>Hue Circular Std</Text>
            <Text style={styles.text}>{data.hue_circular_std}</Text>
            <Text style={styles.text_bold}>Hue Median</Text>
            <Text style={styles.text}>{data.hue_median}</Text>
          </View>
        ) : (
          <View style={styles.spinner}>
            <Spinner size="large" />
            <Text style={styles.spinnertext}>
              Die Ergebnisse werden geladen ...
            </Text>
          </View>
        )}
      </Layout>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'column',
    padding: 40,
  },
  preview: {
    flex: 1,
    justifyContent: 'center',
  },
  button: {
    margin: 20,
  },
  spinner: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  spinnertext: {
    color: 'white',
    margin: 20,
  },
  text: {
    marginBottom: 10,
    color: 'white',
  },
  text_bold: {
    fontWeight: '700',
    color: 'white',
  },
  image_container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
  },
  avatar: {
    height: 200,
    width: 200,
    borderRadius: 200,
    margin: 8,
  },
});
