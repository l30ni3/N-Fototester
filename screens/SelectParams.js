import React, {useState, useRef, useEffect} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  Keyboard,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';
import {
  Button,
  Divider,
  Icon,
  IndexPath,
  Input,
  Layout,
  TopNavigation,
  TopNavigationAction,
  Select,
  SelectItem,
  Text,
} from '@ui-kitten/components';

const GLOBAL = require('./components/constants');

export const SelectParams = ({navigation}) => {
  const [crop, setCrop] = useState(new IndexPath(0));
  const [growth, setGrowth] = useState(new IndexPath(0));
  const [variant, setVariant] = React.useState('');
  const [replicate, setReplicate] = React.useState('');

  const displayCropValue = GLOBAL.CROP_DATA[crop.row];
  const displayGrowthValue = GLOBAL.GROWTH_DATA[growth.row];

  const BackIcon = props => <Icon {...props} name="chevron-left-outline" />;

  const navigateBack = () => {
    navigation.goBack();
  };

  const BackAction = () => (
    <TopNavigationAction icon={BackIcon} onPress={navigateBack} />
  );

  const renderOption = title => <SelectItem title={title} key={title} />;

  const handleVariantText = nextValue => {
    setVariant(nextValue);
  };

  const handleReplicateText = nextValue => {
    setReplicate(nextValue);
  };
  return (
    <SafeAreaView style={{flex: 1, backgroundColor: '#101426'}}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={styles.keyboard_wrapper}
        keyboardVerticalOffset={100}>
        <TopNavigation
          title="Neue Messung"
          alignment="center"
          accessoryLeft={BackAction}
        />
        <Divider />

        <Layout style={styles.container} level="4">
          <Layout style={styles.form}>
            <Text style={styles.form_title} category="h6">
              Was möchten Sie analysieren?
            </Text>
            <Text category="s1">
              Geben Sie mehr Details zur Pflanze an, um später Ihre Ergebnisse
              besser vergleichen zu können.
            </Text>
            <Select
              label={evaProps => (
                <Text {...evaProps} style={[evaProps.style, styles.form_label]}>
                  Kultur
                </Text>
              )}
              style={styles.form_select}
              placeholder="Default"
              value={displayCropValue}
              selectedIndex={crop}
              onSelect={index => setCrop(index)}>
              {GLOBAL.CROP_DATA.map(renderOption)}
            </Select>
            <Select
              label={evaProps => (
                <Text {...evaProps} style={[evaProps.style, styles.form_label]}>
                  Entwicklungsstadium
                </Text>
              )}
              style={styles.form_select}
              placeholder="Default"
              value={displayGrowthValue}
              selectedIndex={growth}
              onSelect={index => setGrowth(index)}
              testID="select-growth">
              {GLOBAL.GROWTH_DATA.map(renderOption)}
            </Select>
            <Input
              label={evaProps => (
                <Text {...evaProps} style={[evaProps.style, styles.form_label]}>
                  Variante
                </Text>
              )}
              placeholder="Bezeichnung eingeben"
              value={variant}
              onChangeText={nextValue => handleVariantText(nextValue)}
              testID="input-variant"
            />
            <Input
              label={evaProps => (
                <Text {...evaProps} style={[evaProps.style, styles.form_label]}>
                  Wiederholung
                </Text>
              )}
              placeholder="Bezeichnung eingeben"
              value={replicate}
              onChangeText={nextValue => handleReplicateText(nextValue)}
              testID="input-replicate"
            />

            <Button
              style={styles.form_submit}
              onPress={() => {
                Keyboard.dismiss(),
                  navigation.navigate('Neue Messung', {
                    crop: displayCropValue,
                    growth: displayGrowthValue,
                    variant: variant,
                    replicate: replicate,
                  });
              }}>
              Weiter
            </Button>
          </Layout>
        </Layout>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  keyboard_wrapper: {
    flex: 1,
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    padding: 30,
    backgroundColor: '#101426',
  },
  form: {
    padding: 20,
    borderRadius: 5,
  },
  form_title: {
    paddingTop: 15,
    paddingBottom: 10,
  },
  form_label: {
    paddingTop: 15,
    paddingBottom: 5,
  },
  form_select: {
    paddingTop: 5,
    paddingBottom: 5,
  },
  form_submit: {
    marginTop: 20,
    marginBottom: 5,
  },
});
