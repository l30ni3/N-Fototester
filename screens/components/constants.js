module.exports = {
  SERVER_URL: 'http://localhost:5000',
  CROP_DATA: ['Winterweizen', 'Sommerweizen', 'Mais'],
  VARIETY_DATA: ['Akteur', 'Alexander', 'Alfons', 'Alpha'],
  GROWTH_DATA: [
    'BBCH 20',
    'BBCH 21',
    'BBCH 22',
    'BBCH 23',
    'BBCH 24',
    'BBCH 25',
    'BBCH 26',
    'BBCH 27',
    'BBCH 28',
    'BBCH 29',
  ],
  SOWING_DATA: ['früh', 'normal', 'spät'],
  CAROUSEL_ITEMS: [
    {
      title: 'Item 1',
      text: 'Text 1',
    },
    {
      title: 'Item 2',
      text: 'Text 2',
    },
    {
      title: 'Item 3',
      text: 'Text 3',
    },
    {
      title: 'Item 4',
      text: 'Text 4',
    },
    {
      title: 'Item 5',
      text: 'Text 5',
    },
  ],
};
