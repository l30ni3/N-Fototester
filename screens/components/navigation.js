import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createDrawerNavigator} from '@react-navigation/drawer';
import {
  Drawer,
  DrawerItem,
  IndexPath,
  DrawerGroup,
  Icon,
  Text,
} from '@ui-kitten/components';
import {HomeScreen} from '../HomeScreen';
import {SelectParams} from '../SelectParams';
import {ImageUpload} from '../ImageUpload';
import {ResultScreen} from '../ResultScreen';

const {Navigator, Screen} = createDrawerNavigator();

const SmartphoneIcon = props => <Icon {...props} name="smartphone-outline" />;

const BrowserIcon = props => <Icon {...props} name="browser-outline" />;

const ColorPaletteIcon = props => (
  <Icon {...props} name="color-palette-outline" />
);

const StarIcon = props => <Icon {...props} name="star" />;

export const DrawerGroupsShowcase = () => {
  const [selectedIndex, setSelectedIndex] = React.useState(null);

  return (
    <Drawer
      selectedIndex={selectedIndex}
      onSelect={index => setSelectedIndex(index)}>
      <DrawerGroup title="Messungen" accessoryLeft={SmartphoneIcon}>
        <DrawerItem title="Meine Messungen" accessoryLeft={StarIcon} />
        <DrawerItem title="Neue Messung" accessoryLeft={StarIcon} />
      </DrawerGroup>
      <DrawerGroup title="Rechtliches" accessoryLeft={ColorPaletteIcon}>
        <DrawerItem title="Impressum" accessoryLeft={StarIcon} />
        <DrawerItem title="FAQs" accessoryLeft={StarIcon} />
      </DrawerGroup>
      <DrawerGroup title="Mein Account" accessoryLeft={BrowserIcon}>
        <DrawerItem title="Settings" accessoryLeft={StarIcon} />
        <DrawerItem title="Logout" accessoryLeft={StarIcon} />
      </DrawerGroup>
    </Drawer>
  );
};

const DrawerContent = ({navigation, state}) => (
  <Drawer
    selectedIndex={new IndexPath(state.index)}
    onSelect={index => navigation.navigate(state.routeNames[index.row])}>
    <DrawerItem title="Alle Messungen" />
    <DrawerItem title="Neue Messung" />
  </Drawer>
);

export const DrawerNavigator = () => (
  <Navigator drawerContent={props => <DrawerContent {...props} />}>
    <Screen name="Meine Messungen" component={HomeScreen} />
    <Screen name="Optionen wÃ¤hlen" component={SelectParams} />
    <Screen name="Neue Messung" component={ImageUpload} />
    <Screen name="Ergebnis" component={ResultScreen} />
  </Navigator>
);

export const AppNavigator = () => (
  <NavigationContainer>
    <DrawerNavigator />
  </NavigationContainer>
);
