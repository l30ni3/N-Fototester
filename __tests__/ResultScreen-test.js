/**
 * @format
 */

import 'react-native';
import React from 'react';
import {ResultScreen} from '../screens/ResultScreen';
import * as eva from '@eva-design/eva';
import {EvaIconsPack} from '@ui-kitten/eva-icons';
import {ApplicationProvider, IconRegistry} from '@ui-kitten/components';
import renderer from 'react-test-renderer';
jest.useFakeTimers();
jest.mock('@react-navigation/native');

const mockedParams = {
  route: {
    params: {
      itemId: 1,
    },
  },
  navigation: '',
};

it('renders correctly', () => {
  const tree = renderer
    .create(
      <>
        <IconRegistry icons={EvaIconsPack} />
        <ApplicationProvider {...eva} theme={eva.light}>
          <ResultScreen {...mockedParams} />
        </ApplicationProvider>
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
