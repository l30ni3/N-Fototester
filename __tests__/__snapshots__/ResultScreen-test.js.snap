// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "minHeight": 56,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "space-between",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "zIndex": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="#222B45"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": 24,
                    "tintColor": "#222B45",
                    "width": 24,
                  },
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                fill={
                  Array [
                    0,
                    4280429381,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                font={Object {}}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        0,
                        4278190080,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  >
                    <RNSVGRect
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      height="24"
                      matrix={
                        Array [
                          6.123233995736766e-17,
                          1,
                          -1,
                          6.123233995736766e-17,
                          24,
                          0,
                        ]
                      }
                      opacity={0}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      width="24"
                      x={0}
                      y={0}
                    />
                    <RNSVGPath
                      d="M13.36 17a1 1 0 0 1-.72-.31l-3.86-4a1 1 0 0 1 0-1.4l4-4a1 1 0 1 1 1.42 1.42L10.9 12l3.18 3.3a1 1 0 0 1 0 1.41 1 1 0 0 1-.72.29z"
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          Ergebnis
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "zIndex": 1,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#EDF1F7",
            "height": 1,
          },
          undefined,
        ]
      }
    />
    <View
      appearance="default"
      level="4"
      style={
        Array [
          Object {
            "backgroundColor": "#E4E9F2",
          },
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "",
            }
          }
          style={
            Object {
              "height": 333.5,
              "width": 750,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "flex-start",
            },
            Object {
              "height": 1000.5,
              "width": 750,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#3366FF",
              "fontSize": 120,
              "padding": 10,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "700",
            }
          }
        >
          Datum:
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "marginBottom": 10,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "700",
            }
          }
        >
          Kultur:
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "marginBottom": 10,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "700",
            }
          }
        >
          Entwicklungsstadium:
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "marginBottom": 10,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "700",
            }
          }
        >
          Variantenbezeichnung:
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "marginBottom": 10,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "700",
            }
          }
        >
          Wiederholung:
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "marginBottom": 10,
            }
          }
        />
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
